tools/test-backend

```diff
@@ -66,6 +71,56 @@ if __name__ == "__main__":
                       help="Show which tests are slowest.")
 
     (options, args) = parser.parse_args()
+
+    thedir = 'zerver/tests/'
+
+    for suite in args:
+        args[args.index(suite)] = suite.replace("/", ".")
+
+    for suite in args:
+        found_file = False
+        if suite[0].isupper() and "test_" in suite:
+            classname = suite.rsplit('.', 1)[0]
+            for file in os.listdir(thedir):
+                if not file.endswith(".pyc"):
+                    filepath = thedir + file
+                    if not os.path.isdir(filepath):
+                        for line in open(filepath):
+                            if classname in line:
+                                filepath = filepath.replace(".py", ".")
+                                new_suite = filepath + suite
+                                args[args.index(suite)] = new_suite
+                                found_file = True
+                                break
+                if found_file == True:
+                    break
+
+        elif suite[0].isupper():
+            for file in os.listdir(thedir):
+                if not file.endswith(".pyc"):
+                    filepath = thedir + file
+                    if not os.path.isdir(filepath):
+                        for line in open(filepath):
+                            if suite in line:
+                                filepath = filepath.replace(".py", ".")
+                                new_suite = filepath + suite
+                                args[args.index(suite)] = new_suite
+                                found_file = True
+                                break
+                if found_file == True:
+                    break
+
+    for suite in args:
+        if suite.startswith('test'):
+            new_suite = "zerver.tests." + suite
+            args[args.index(suite)] = new_suite
+
+    for suite in args:
+        args[args.index(suite)] = suite.replace(".py", "")
+
+    for suite in args:
+        args[args.index(suite)] = suite.replace("/", ".")
```

Comment:
Yes. Let's suppose someone entered test_module/test_class. So the first transformation will convert it to test_module.test_class as further code runs considering dots (searching and splitting based on dots).
But then later I have defined thedir = "zerver/test" which is important since the path has to with forward slashes, I do the same transformation in the end, when we get the exact address.
Its for allowing user to have test_module/test_class or test_class/test_single along with test_module.test_class and so on.